packages/rocketchat-lib/server/lib/sendNotificationsOnMessage.js

```diff
@@ -206,9 +206,9 @@ RocketChat.callbacks.add('afterSaveMessage', function(message, room, userId) {
 	subscriptions.forEach((s) => {
 		userIds.push(s.u._id);
 	});
-	const userSettings = {};
-	RocketChat.models.Users.findUsersByIds(userIds, { fields: { 'settings.preferences.audioNotifications': 1, 'settings.preferences.desktopNotifications': 1, 'settings.preferences.mobileNotifications': 1 } }).forEach((user) => {
-		userSettings[user._id] = user.settings;
+	const users = {};
+	RocketChat.models.Users.findUsersByIds(userIds, { fields: { 'settings.preferences': 1 } }).forEach((user) => {
```

Comment:
This part is not good, we was getting only users that did change their configurations, not all users as this code does now.

Reply:
@rodrigok Actually, as I didn't touch the `userIds` list and we are using a findByIds function, I think we are getting the same users, but now returning all their preferences. Or the `fields` argument generates something like a `where` clause?

Reply:
🤦‍♂️ You're right, was too much code to read 😄